geode-core/src/main/java/org/apache/geode/internal/cache/tier/sockets/ServerConnectionFactory.java

```diff
@@ -72,9 +99,15 @@ public static ServerConnection makeServerConnection(Socket s, InternalCache c,
         throw new IOException("Acceptor received unknown communication mode: " + communicationMode);
       } else {
         protobufProtocolHandler = findClientProtocolMessageHandler();
-        return new GenericProtocolServerConnection(s, c, helper, stats, hsTimeout, socketBufferSize,
-            communicationModeStr, communicationMode, acceptor, protobufProtocolHandler,
-            securityService);
+        authenticatorClass = findStreamAuthenticator(
+            c.getInternalDistributedSystem().getConfig().getProtobufProtocolAuthenticationMode());
+        try {
+          return new GenericProtocolServerConnection(s, c, helper, stats, hsTimeout,
+              socketBufferSize, communicationModeStr, communicationMode, acceptor,
+              protobufProtocolHandler, securityService, authenticatorClass.newInstance());
```

Comment:
We find the class exactly once, and create a new instance per protocol instance. We then create an instance per `GenericProtocolServerConnection`. This is because the authenticator (unlike the protocol handler) is stateful, so we can't share it between connections.